drivers/sensor/lis2dux12/lis2dux12.h

```diff
@@ -0,0 +1,91 @@
+/* ST Microelectronics LIS2DUX12 3-axis accelerometer driver
+ *
+ * Copyright (c) 2023 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Datasheet:
+ * https://www.st.com/resource/en/datasheet/lis2dux12.pdf
+ */
+
+#ifndef ZEPHYR_DRIVERS_SENSOR_LIS2DUX12_LIS2DUX12_H_
+#define ZEPHYR_DRIVERS_SENSOR_LIS2DUX12_LIS2DUX12_H_
+
+#include <zephyr/types.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/sensor/stmemsc.h>
+#include "lis2dux12_reg.h"
+
+#if DT_ANY_INST_ON_BUS_STATUS_OKAY(spi)
+#include <zephyr/drivers/spi.h>
+#endif /* DT_ANY_INST_ON_BUS_STATUS_OKAY(spi) */
+
+#if DT_ANY_INST_ON_BUS_STATUS_OKAY(i2c)
+#include <zephyr/drivers/i2c.h>
+#endif /* DT_ANY_INST_ON_BUS_STATUS_OKAY(i2c) */
+
+/* Return ODR reg value based on data rate set */
+#define LIS2DUX12_ODR_TO_REG(_odr) \
+	((_odr <= 1) ? 1 : \
+	((31 - __builtin_clz(_odr / 25))) + 3)
```

Comment:
This one looks to me totally wrong. I tried to set 100Hz and I obtained 12.5 Hz. In lsm6dsv16x (and others) I use a odr_map table. Whatever you may want to use, pls fix it for all cases.

Reply:
This was blindly borrowed from [here](https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/sensor/lis2dw12/lis2dw12.h#L30-L33). I've since replaced it with the approach you mentioned. Thank you.

Reply:
Here is the relevant change:
https://github.com/zephyrproject-rtos/zephyr/blob/621f99f53dc8ed00f1cebb3fcc6f704d1e808c1e/drivers/sensor/lis2dux12/lis2dux12.c#L68-L104